### DISCUSSION TITLE
Adding VNET and NAT Gateway to get static outbound IP

### DISCUSSION THREAD

**Discussion by PleaseStopAsking** [polymind-inc/acmebot#585] [Ideas]
Hi all,

I am currently doing some testing with this project and have a couple requirements around my environment that would likely apply to others as well. Specifically, I am utilizing Route 53 for my DNS provider but have Service Control Policies implemented at the AWS Org level which restricts access to my AWS accounts to only certain IP ranges which obviously causes issues with `keyvault-acmebot`.

The solution here is to move to a premium sku function and integrate with a vnet that has a Nat Gateway. This would still default to consumption sku but allow the premium sku if the parameters are provided.

I want to get an idea if this is something that would be accepted as a PR if I contributed the changes as well as start a discussion.

**Comment by shibayan** [comment]
It is difficult to accept a PR that is created to be deployed outside of the Consumption Plan. It is very costly and it may be cheaper to purchase a certificate.

However, we welcome blogs or other means to show how to deploy.
